packages/stateful/actions/components/AuthzAuthorization.tsx

```diff
@@ -0,0 +1,124 @@
+  const { register, setValue, watch } = useFormContext()
+
+  return (
+    <ActionCard
+      Icon={KeyEmoji}
+      onRemove={onRemove}
+      title={t('title.authzAuthorization')}
+    >
+      <div className="flex flex-col items-stretch gap-1">
+        <InputLabel name={t('form.grantOrRevokeAuthz')} />
+        <SelectInput
+          disabled={!isCreating}
+          fieldName={fieldNamePrefix + 'type_url'}
+          register={register}
+        >
+          <option value="/cosmos.authz.v1beta1.MsgGrant">
+            {t('form.grantAuthorizationOption')}
+          </option>
+          <option value="/cosmos.authz.v1beta1.MsgRevoke">
+            {t('form.revokeAuthorizationOption')}
+          </option>
+        </SelectInput>
+      </div>
+
+      <div className="flex flex-col items-stretch gap-1">
+        <InputLabel
+          name={t('form.granteeAddress')}
+          tooltip={t('form.granteeAddressTooltip')}
+        />
+        <TextInput
+          disabled={!isCreating}
+          error={errors?.value?.grantee}
+          fieldName={fieldNamePrefix + 'value.grantee'}
+          placeholder={!isCreating ? t('info.none') : 'juno...'}
+          register={register}
+          validation={[
+            (v: string) =>
+              validateAddress(v) || validateContractAddress(v, false),
+          ]}
+        />
+        <InputErrorMessage error={errors?.admin} />
+      </div>
+
+      {!data.custom ? (
+        <div className="flex flex-col items-stretch gap-1">
+          <InputLabel name={t('form.messageType')} />
+          <SelectInput
+            disabled={!isCreating}
+            fieldName={fieldNamePrefix + 'value.msgTypeUrl'}
+            register={register}
+          >
+            <option value={AuthzExecActionTypes.Delegate}>
+              {t('info.stake')}
+            </option>
+            <option value={AuthzExecActionTypes.Undelegate}>
+              {t('info.unstake')}
+            </option>
+            <option value={AuthzExecActionTypes.Redelegate}>
+              {t('info.redelegate')}
+            </option>
+            <option value={AuthzExecActionTypes.ClaimRewards}>
+              {t('info.withdrawStakingRewards')}
+            </option>
+          </SelectInput>
+        </div>
+      ) : (
+        <div className="flex flex-col items-stretch gap-1">
+          <InputLabel name={t('form.messageType')} />
+          <TextInput
+            disabled={!isCreating}
+            error={errors?.value?.msgTypeUrl}
+            fieldName={fieldNamePrefix + 'value.msgTypeUrl'}
+            placeholder={!isCreating ? t('info.none') : t('form.messageType')}
+            register={register}
+            validation={[(v: string) => validateRequired(v)]}
+          />
+          <InputErrorMessage error={errors?.value?.msgTypeUrl} />
+        </div>
+      )}
+
+      <div className="bg-card flex grow flex-row items-center justify-between gap-4 rounded-md py-2">
```

Comment:
this container is no longer needed; `bg-card` doesn't exist, but if it did, there would be a weird background padding container to this switch card that the switch card already handles itself. after removing this, it might be worth adding `mt-2` to the `FormSwitchCard`'s `containerClassName` prop so preserve the spacing.